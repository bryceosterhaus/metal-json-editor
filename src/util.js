export const TYPE_COLORS = {
	boolean: '#AA0D91',
	null: '#FF8C00',
	number: '#1C00CF',
	readOnly: '#117700',
	string: '#C41A16'
};

export function setIn(initVal, locator = [], value) {
	if (!(initVal instanceof Object)) {
		return initVal;
	}

	let clonedVal = JSON.parse(JSON.stringify(initVal));

	let retVal = clonedVal;

	let i;

	for (i = 0; i < locator.length - 1; i++) {
		retVal = retVal[locator[i]];
	}

	retVal[locator[i]] = value;

	return clonedVal;
}