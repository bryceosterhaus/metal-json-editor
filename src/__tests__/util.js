import {setIn} from '../util';

describe('setIn', () => {
  it('should set value in object', () => {
    expect(setIn({foo: ''}, ['foo'], 'bar')).toMatchSnapshot();
  });

  it('should set nested value in object', () => {
    expect(setIn({foo: {bar: ''}}, ['foo', 'bar'], 'baz')).toMatchSnapshot();
  });

  it('should set value in array', () => {
    expect(setIn(['foo'], [0], 'baz')).toMatchSnapshot();
  });

  it('should return same value if not an object instance', () => {
    expect(setIn(1, [0], 'bar')).toMatchSnapshot();
  });
});
