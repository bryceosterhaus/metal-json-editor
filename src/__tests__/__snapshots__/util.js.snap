// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setIn should return same value if not an object instance 1`] = `1`;

exports[`setIn should set nested value in object 1`] = `
Object {
  "foo": Object {
    "bar": "baz",
  },
}
`;

exports[`setIn should set value in array 1`] = `
Array [
  "baz",
]
`;

exports[`setIn should set value in object 1`] = `
Object {
  "foo": "bar",
}
`;
